crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -461,9 +495,11 @@ impl AlarmConverter {
         }
     }
 
-    fn try_convert_alarm(&mut self, input: &Message) -> Result<Vec<Message>, ConversionError> {
-        let mut vec: Vec<Message> = Vec::new();
-
+    fn try_convert_alarm(
+        &mut self,
+        mqtt_messages: &mut Vec<Message>,
```

Comment:
I really got confused with the two arguments: `mqtt_messages` and `input` since both are `Message`. I think this `mqtt_messages` should be renamed to `output_messages` because MQTT is very obvious from the type and good contrast with `input`.

```suggestion
        output_messages: &mut Vec<Message>,
```

Reply:
Instead of passing the vector of `output_messages` into this function, I'd rather append the result of the this function to the messages vector in the calling function: `process_alarm_messages`.